lib/extensions/ar_order.rb

```diff
@@ -0,0 +1,38 @@
+# This monkey patches rails to support proper joins
+# This is a current PR for 6.1: https://github.com/rails/rails/pull/36531
+module ActiveRecord
+  module ConnectionAdapters
+    module PostgreSQL
+      module SchemaStatements
+        def columns_for_distinct(columns, orders) #:nodoc:
+          order_columns = orders.reject(&:blank?).map { |s|
+              # Convert Arel node to string
+              unless s.is_a?(String)
+                if s.kind_of?(Arel::Nodes::Ordering)
+                  s = s.expr
+                  keep_order = true
+                end
+                if s.respond_to?(:to_sql)
+                  s = s.to_sql
+                else # for Arel::Nodes::Attribute
+                  engine = Arel::Table.engine
+                  collector = Arel::Collectors::SQLString.new
+                  collector = engine.connection.visitor.accept s, collector
+                  s = collector.value
+                end
+              end
+              # Remove any ASC/DESC modifiers
```

Comment:
This comment should be updated to reflect the conditional you're doing here.  We should explain that there's an existing ORDER clause so we want to retain any ASC/DESC.